protocol/messenger_profile_showcase.go

```diff
@@ -349,8 +400,29 @@ func (m *Messenger) GetProfileShowcasePreferences() (*identity.ProfileShowcasePr
 	return m.persistence.GetProfileShowcasePreferences()
 }
 
-func (m *Messenger) GetProfileShowcaseForContact(contactID string) (*identity.ProfileShowcase, error) {
-	return m.persistence.GetProfileShowcaseForContact(contactID)
+func (m *Messenger) GetProfileShowcaseForContact(contactID string, validate bool) (*identity.ProfileShowcase, error) {
+	profileShowcase, err := m.persistence.GetProfileShowcaseForContact(contactID)
+	if err != nil {
+		return nil, err
+	}
+
+	if !validate {
+		return profileShowcase, nil
+	}
+
+	contactPubKey, err := common.HexToPubkey(contactID)
+	if err != nil {
+		return nil, err
+	}
+
+	profileShowcase.Communities, err = m.validateCommunitiesMembership(profileShowcase.Communities, contactPubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: validate collectibles & assets ownership
```

Comment:
Should we have a ticket?

Reply:
should be part of https://github.com/status-im/status-desktop/issues/14129, added to comment